core/state/statedb.go

```diff
@@ -218,10 +218,14 @@ func (s *StateDB) AddLog(log *types.Log) {
 }
 
 func (s *StateDB) GetLogs(hash common.Hash) []*types.Log {
+	s.mutex.RLock()
```

Comment:
It is generally not best practice to put locks on reads operation of map[]. Reconsider removing this.